### ISSUE TITLE
global destructors

### ISSUE THREAD

**Issue by Trass3r** [cmu-sei/pharos#40]
I saw functions being marked as constructors which are actually destructors.
It could have been deduced from atexit usage: https://godbolt.org/z/rndZOs
(Also in this particular case the function called operator delete which shouldn't happen in a ctor.)

**Comment by Trass3r** [comment]
Nice to have: exported information about these 'dynamic initializers' etc., i.e. "function x is a dynamic atexit destructor for 'foo'", ...

**Comment by sei-ccohen** [comment]
Agreed that this would be nice.  No progress yet.  Leaving open as an outstanding enhancement, but it's also unlikely to get attention for a while.

**Comment by sei-eschwartz** [comment]
@sei-ccohen Is there a reason why we don't have a rule or guess stating if a method calls delete it isn't a constructor?

**Comment by sei-eschwartz** [comment]
@Trass3r We changed a bunch of rules relating to destructor detection recently.  You may want to retry this.
